Gems/XR/Code/Source/XRUtils.cpp

```diff
@@ -32,16 +32,32 @@ namespace XR
         { 
             result.SetRow(0, 2.0f / tanAngleWidth, 0.0f, (left + right) / tanAngleWidth, 0.0f);
             result.SetRow(1, 0.0f, 2.0f / tanAngleHeight, (top + bottom) / tanAngleHeight, 0.0f);
-            result.SetRow(2, 0.0f, 0.0f, -1 * (farDist + nearDist) * invfn, -2.0f * farDist * nearDist * invfn);
+            if(reverseDepth)
+            { 
+                result.SetRow(2, 0.0f, 0.0f, 2.0f * nearDist * invfn, 2.0f * farDist * nearDist * invfn);
+            }
+            else
+            {
+                result.SetRow(2, 0.0f, 0.0f, -1.0f * (farDist + nearDist) * invfn, -2.0f * farDist * nearDist * invfn);
+            }
             result.SetRow(3, 0.0f, 0.0f, -1.0f, 0.0f);
         }
         else
         {
+            // place the far plane at infinity
```

Comment:
Why set far plane to infinity if far dist is smaller than near dist?

Reply:
In theory you can use that to render sky, sun, moon, stars, etc. It basically takes f to infinity so -(f+n)/(f-n) essentially becomes -1. It has the effect of mapping vertex positions on the infinity distant far plane. You could also use it to render stencil shadow volumes too. But given that the non-stereoscopic projection function doesnt support this functionality, it wont be used for stereoscopic projections as well. Regardless let me update the comments in the header to make this clear.